.github/actions/verify-token/src/main.ts

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2022 SLSA Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WIHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as core from "@actions/core";
+import * as sigstore from "sigstore";
+import * as process from "process";
+import * as fs from "fs";
+import * as child_process from "child_process";
+
+interface githubObj {
+  event_name: string;
+  run_attempt: string;
+  run_id: string;
+  run_number: string;
+  workflow: string;
+  sha: string;
+  repository: string;
+  repository_owner: string;
+  // TODO(#1411): Record if these become available.
+  // repository_id: string;
+  // repository_owner_id: string;
+  // actor_id: string;
+  ref: string;
+  ref_type: string;
+  actor: string;
+}
+
+async function run(): Promise<void> {
+  try {
+    /* Test locally:
+        $ env INPUT_SLSA-WORKFLOW-RECIPIENT="delegator_generic_slsa3.yml" \
+        INPUT_SLSA-UNVERIFIED-TOKEN="$(cat testdata/slsa-token)" \
+        GITHUB_EVENT_NAME="workflow_dispatch" \
+        GITHUB_RUN_ATTEMPT="1" \
+        GITHUB_RUN_ID="3789839403" \
+        GITHUB_RUN_NUMBER="198" \
+        GITHUB_WORKFLOW="delegate release project" \
+        GITHUB_SHA="8cbf4d422367d8499d5980a837cb9cc8e1e67001" \
+        GITHUB_REPOSITORY="laurentsimon/slsa-delegate-project" \
+        GITHUB_REPOSITORY_OWNER="laurentsimon" \
+        GITHUB_REF="refs/heads/main" \
+        GITHUB_REF_TYPE="branch" \
+        GITHUB_ACTOR="laurentsimon" \
+        nodejs ./dist/index.js
+    */
+
+    const workflowRecipient = core.getInput("slsa-workflow-recipient");
+    const unverifiedToken = core.getInput("slsa-unverified-token");
+
+    // Log the inputs for troubleshooting.
+    core.debug(`workflowRecipient: ${workflowRecipient}`);
+    core.debug(`unverifiedToken: ${unverifiedToken}`);
+    const parts = unverifiedToken.split(".");
+    if (parts.length !== 2) {
+      throw new Error(`malformed token: ${parts.length}`);
+    }
+
+    const bundleStr = Buffer.from(parts[0], "base64").toString("utf-8");
+    const b64Token = parts[1];
+    const bundle = JSON.parse(bundleStr);
+
+    // First, verify the signature, i.e., that it is signed by a certificate that
+    // chains up to Fulcio.
+    await sigstore.sigstore.verify(bundle, Buffer.from(b64Token));
```

Comment:
so the payload signed is the base64-encoded token? why not the rawToken?

Reply:
1. less parsing before verification, so less attack surface
2. base64 decoding allows some bits to be flipped without changing the decoded results, which seems weird for signature verification to still pass

Not a big deal, I agree :)